### PR TITLE
TST: remove invalid internals tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Delay validation check in get_slice is also nice.

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -711,16 +711,16 @@ def combine(self, blocks, copy=True):
...
         if axis == 0:
             new_blocks = self._slice_take_blocks_ax0(slobj)
-        else:
+        elif axis == 1:
             _slicer = [slice(None)] * (axis + 1)
             _slicer[axis] = slobj
             slicer = tuple(_slicer)
             new_blocks = [blk.getitem_block(slicer) for blk in self.blocks]
+        else:
+            raise IndexError("Requested axis not found in manager")
```
Comment (simonjayhawkins) [inline pandas/core/internals/managers.py]: self.ndim==1 and axis==1 are guarded against elsewhere?

Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: yes

**Review by WillAyd — APPROVED** [review]
Lgtm

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**